core/platform/auth/firebase_auth_services.py

```diff
@@ -196,18 +198,25 @@ def associate_multi_auth_ids_to_user_ids(pairs):
     # Turn list(pair) to pair(list): https://stackoverflow.com/a/7558990/4859885
     auth_ids, user_ids = python_utils.ZIP(*pairs)
 
-    claimed_user_ids = get_multi_user_ids_from_auth_ids(auth_ids)
-    if any(user_id is not None for user_id in claimed_user_ids):
-        existing_associations = sorted(
-            'auth_id=%r, user_id=%r' % (auth_id, user_id)
-            for auth_id, user_id in python_utils.ZIP(auth_ids, claimed_user_ids)
-            if user_id is not None)
-        raise Exception(
-            'associations already exist for: %r' % (existing_associations,))
-
-    mappings = [
-        auth_models.UserIdByFirebaseAuthIdModel(id=auth_id, user_id=user_id)
-        for auth_id, user_id in python_utils.ZIP(auth_ids, user_ids)
-    ]
-    auth_models.UserIdByFirebaseAuthIdModel.update_timestamps_multi(mappings)
-    auth_models.UserIdByFirebaseAuthIdModel.put_multi(mappings)
+    def _associate_multi_auth_ids_to_user_ids_transactionally():
```

Comment:
Suggest using our new transactional decorator, instead:

https://github.com/oppia/oppia/blob/b46178fc512402303e1144c8b5590f45af5ce7b3/core/platform/transactions/gae_transaction_services.py#L55

This conveniently helps us avoid creating an extra function to run in transaction.

Reply:
Done!